### PR TITLE
Table view: move layout elements from component to view

### PR DESCRIPTION
**Pull request by MetRonnie**
Best viewed with whitespace hidden

### CODE REVIEW

File: src/components/cylc/table/Table.vue [hunk 1]
```diff
@@ -16,102 +16,87 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+        size="small"
+        :style="{
+          visibility: (item.task.children || []).length ? null : 'hidden',
+          transform: isExpanded(internalItem) ? 'rotate(180deg)' : null
+        }"
       >
-        <v-container
-          fluid
-          class="ma-0 pa-0 w-100 h-100 left-0 top-0 position-absolute pt-2"
-        >
```
Comment (MetRonnie) [inline src/components/cylc/table/Table.vue:30]: Moved these to the view too

File: src/views/Table.vue [hunk 2]
```diff
@@ -167,16 +176,10 @@ export default {
      * The job id input and selected task filter state.
      * @type {import('vue').Ref<object>}
      */
-    const tasksFilter = useInitialOptions('tasksFilter', { props, emit }, ref({}))
-    /**
-     * The state for toggling whether the filters are rendered
-     * @type {import('vue').Ref<boolean>}
-     */
-    const filterable = useInitialOptions('filterable', { props, emit }, ref(true))
```
Comment (MetRonnie) [inline src/views/Table.vue:175]: No need for `filterable` anymore, the filters are always shown in the view

File: src/views/Tree.vue [hunk 3]
```diff
@@ -228,7 +227,7 @@ export default {
      * The job id input and selected task filter state.
      * @type {import('vue').Ref<Object>}
      */
-    const tasksFilter = useInitialOptions('tasksFilter', { props, emit }, ref({ id: null, states: null }))
+    const tasksFilter = useInitialOptions('tasksFilter', { props, emit }, { id: null, states: null })
```
Comment (MetRonnie) [inline src/views/Tree.vue:230]: N.B. the default value should be an non-reactive variable

File: tests/unit/views/table.vue.spec.js [hunk 4]
```diff
@@ -81,26 +77,19 @@ describe('Table view', () => {
...
         workflowName: 'one',
       },
-      data: () => ({
-        // Override computed property
-        workflows
-      })
     })
 
+    await wrapper.setData({ workflows })
+
```
Comment (MetRonnie) [inline tests/unit/views/table.vue.spec.js:92]: This fixes some warnings that were showing up when running the tests

**Review by markgrahamdawson — APPROVED** [review]

**Pull request merged**